Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarState.kt

```diff
@@ -0,0 +1,26 @@
+package android.clicker.school_live_simulator
+
+/**
+ * @author  Nikita Dimitryuk niksvob@mail.ru
+ */
+abstract class GuitarState {
+    protected abstract val price: Int
+    protected val available_courses: ArrayList<String> = arrayListOf()
+
+    /**
+     * Updates the state of the purchased guitar.
+     *
+     * @param    bag player inventory
+     */
+    abstract fun changeState(bag: Player.Bag)
+
+    /**
+     * Function checks if the course is available
+     *
+     * @param    courseName  name of the course
+     * @return               course availability (true/false)
+     */
+    open fun isAvailable(courseName: String): Boolean {
+        return available_courses.contains(courseName)
```

Comment:
@Ferrociraptus  Is that better than using 'in'?

Reply:
It is not bad, because it is Java like stile (operator "in" work only in Kotlin lang).
Operator "in" is equivalent to contains method.

Reply:
> It is not bad, because it is Java like stile (operator "in" work only in Kotlin lang).
> Operator "in" is equivalent to contains method.

I understand. I think, that these two solutions are similar but we have to choose one because Andrey uses 'in'. If we like 'in'  (or 'contains') more we should use it in every class..

Reply:
Okay, I'll fix it for operator "in" tonight

Reply:
> Okay, I'll fix it for operator "in" tonight

You don't have to fix something until Arseny says his opininion about better solution. Maybe Andrey has to use contains instead of in

Reply:
@Bazalii, I think, that all is fine. Both variants are good to read and use in code. In my opinion, it can be written different in all code.

Reply:
> @Bazalii, I think, that all is fine. Both variants are good to read and use in code. In my opinion, it can be written different in all code.

Okay, lets have two variants in our code!